### ISSUE TITLE
atom-shell

### ISSUE THREAD

**Issue by daiyam** [mapbox/node-sqlite3#357]
Hi,

How can I compile node-sqlite3 for _atom-shell_? I want to use it in the web-page side to avoid any unnecessary ipc call.

Thx

**Comment by jasonhinkle** [comment]
I was able to get sqlite3 working with Electron after install with the following.  Note that the version numbers are hard-coded into the arguments which are Electron 0.27.1 and v43 (by the way anybody know where 43 come from? It seems to refer to modules versioning that is related to the version of Electron, but I can't find any translation of these numbers - the only way to find it is to just let your app throw an error and see what the foldername is that it's looking for.)

```bash
cd node_modules/sqlite3
npm run prepublish
node-gyp configure --module_name=node_sqlite3 --module_path=../lib/binding/node-v43-darwin-x64
node-gyp rebuild --target=0.27.1 --arch=x64 --target_platform=darwin --dist-url=https://atom.io/download/atom-shell --module_name=node_sqlite3 --module_path=../lib/binding/node-v43-darwin-x64
```

**Comment by maxogden** [comment]
43 is the NODE_MODULE_VERSION which is similar to the ABI version, node, iojs and electron all have their non-overlapping own version ranges that they increment whenever the native API changes. I don't think this is written down anywhere (i've been trying to get core people to document it)

**Comment by ekryski** [comment]
Gonna also throw in my hat. Would love to see this work out of the box. I can confirm that @jasonhinkle's  solution does work.

I still haven't tried with ASPM as documented [here](https://github.com/atom/electron/issues/1031). I tried with `electron-build` but it didn't seem to work as my require statement still bombed but I didn't dig any deeper there.

**Comment by olimsaidov** [comment]
For anyone having same issue: http://verysimple.com/2015/05/30/using-node_sqlite3-with-electron/

**Comment by ekryski** [comment]
@olimsaidov Nice! I should have written something up myself. I have the exact same process going. Works perfectly.

**Comment by thefinn93** [comment]
Lots of these have the word "darwin" in them (and aren't working for me), are they mac-specific?

**Comment by ekryski** [comment]
@thefinn93 Yes. You need to specify which platform you are targeting. I haven't built for other platforms just yet, so I'm not entirely sure what options you need to change.

**Comment by jasonhinkle** [comment]
The target_platform and target_arch need to be changed for your environment - the options can be found at https://www.npmjs.com/package/node-pre-gyp

**Comment by zcbenz** [comment]
Refs https://github.com/mapbox/node-pre-gyp/pull/175.

**Comment by springmeyer** [comment]
@zcbenz - thank you for mapbox/node-pre-gyp#175 - it is now merged and part of node-pre-gyp v0.6.12. Would it be possible for you to create a pull that starts building binaries for electron on travis?
